core/src/main/java/com/linecorp/armeria/server/encoding/HttpEncodedResponse.java

```diff
@@ -92,7 +101,7 @@ protected HttpObject filter(HttpObject obj) {
                 return obj;
             }
 
-            encodedStream = new ByteArrayOutputStream();
+            encodedStream = new ByteBufOutputStream(alloc.buffer());
```

Comment:
How about specifying the initial capacity using the value from content-length header?

Reply:
I changed to set `content-length / 2`. A compression ratio would be higher than 50%.